src/vm/i386/cgenx86.cpp

```diff
@@ -1607,14 +1607,19 @@ void UMEntryThunkCode::Encode(BYTE* pTargetCode, void* pvSecretParam)
     m_jmp        = X86_INSTR_JMP_REL32;
     m_execstub   = (BYTE*) ((pTargetCode) - (4+((BYTE*)&m_execstub)));
 
-    FlushInstructionCache(GetCurrentProcess(),GetEntryPoint(),sizeof(UMEntryThunkCode));
```

Comment:
Could you please keep the full `FlushInstructionCache` in Encode (on x86 and x64 at least)?

We used to have issues with [Time Travel Debugging](https://docs.microsoft.com/en-us/windows-hardware/drivers/debugger/time-travel-debugging-overview) that required explicit FlushInstructionCache to fix/workaround. I am not sure whether these issues still exist.

Using `ClrFlushInstructionCache` in Poison should be fine.